sparse/test/fiemap_test.go

```diff
@@ -0,0 +1,337 @@
+	err := SyncFile(srcPath, localhost+":"+port, timeout, true)
+	if err != nil {
+		t.Fatalf("sync error: %v", err)
+	}
+	log.Infof("Syncing done, size: %v elapsed: %.2fs", testFileSize, time.Now().Sub(startTime).Seconds())
+
+	startTime = time.Now()
+	log.Info("Checking...")
+	err = checkSparseFiles(srcPath, dstPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	log.Infof("Checking done, size: %v elapsed: %.2fs", testFileSize, time.Now().Sub(startTime).Seconds())
+}
+
+// 200GB creation - 14 minutes
+// 200GB count retrieval - 19.4s
+// 200GB extent retrieval - 9s
+// 200GB check time - 28.5s - 1.4GB memory - 26214400 extents
+// 200GB iterative check time - 21s - 53KB memory - 26214400 extents
+// slow down 316GB written 316GB holes in 1:30h
+// 1000GB creation - slows down after 300GB from (3s per GB to -> 15s per GB)
+// 1000GB iterative retrieval: 0.05s per GB of file - 56byes * 1024extents = 56KB
+// 1000GB total extents 131072000 - 47s processing iteratively
+// 1000GB total extents 131072000 - fails to process completely, probably because it takes 6.5GB of memory
+// 1000GB count retrieval done, time elapsed: 42.37 second, extent number: 131072000
+// total extents count retrieval is unnecessary, processing them iteratively is just as fast
+func TestFileFiemapIterative(t *testing.T) {
+	const (
+		GB = int64(1024 * 1024 * 1024)
+		// testFileName      = srcPrefix + "-fiemap-1000gb-file"
+		// testFileSize      = 1000 * GB
+		testFileName      = srcPrefix + "-fiemap-1gb-file"
+		testFileSize      = 1 * GB
+		testFileBlockSize = int64(4096)
+	)
+
+	t.Skip("skipped fiemap_test::TestFileFiemapIterative")
+	srcPath := filepath.Join(os.TempDir(), testFileName)
+	// srcPath := filepath.Join("/home/jmoody/test-files", testFileName)
+	// srcPath := tempFilePath(testFileName)
+
+	if info, err := os.Stat(srcPath); err != nil || info.Size() == 0 {
+		// in case of error we just create a new test file
+		if err = writeMultipleHolesData(srcPath, testFileSize, testFileBlockSize, testFileBlockSize); err != nil {
+			t.Fatalf("failed to create fiemap test file path: %v error: %v", srcPath, err)
+		}
+	}
+	fileIo, err := NewDirectFileIoProcessor(srcPath, os.O_RDONLY, 0)
+	if err != nil {
+		t.Fatalf("failed opening fiemap test file path: %v error: %v", srcPath, err)
+	}
+	defer fileIo.Close()
+	log.Info("calculating fiemap test file extends")
+	startTime := time.Now()
+
+	// process fiemap iterativly
+	// we can scale this based on the filesize
+	extCount := uint32(1024) // 56 kilobytes
+	extStart := uint64(0)
+	extLength := uint64(testFileSize)
+	// var exts []Extent
+	totalCount := 0
+	fiemap := NewFiemapFile(fileIo.GetFile())
+
+	// retrieve total extent count
+	//count, _, errno := fiemap.FiemapRegion(0, extStart, extLength)
+	//if errno != 0 { t.Fatal("failed to call fiemap.Fiemap(0)") }
+	//log.Infof("count retrieval done, time elapsed: %.2f second, extent number: %v", time.Now().Sub(startTime).Seconds(), count)
+
+	deltaTime := time.Now()
+	for {
```

Comment:
These tests aren't needed anymore, since the new fiemap code is covered by all sync tests.
This is just benchmarking code that can be removed, in the final version.